settled_handler.go

```diff
@@ -64,22 +65,27 @@ func (p *SettledHandler) WaitForInvoiceSettled(ctx context.Context,
 	// Check database to see if invoice was already settled.
 	invoice, _, err := p.persister.Get(ctx, hash)
 	if err != nil {
-		return err
+		return types.StatusUnknown, err
 	}
 	if invoice.ReachFinalStatus {
 		logger.Debugw("Wait for invoice settled completed via db")
 
-		return nil
+		return invoice.Status, nil
 	}
 
 	// Not settled yet. Wait for the event.
 	select {
 	case <-waitChan:
 		logger.Debugw("Wait for invoice settled completed via wait channel")
 
-		return nil
+		invoice, _, err := p.persister.Get(ctx, hash)
```

Comment:
Is this re-fetch necessary or can the final outcome be passed over the `waitChan`?

Reply:
First, I didn;t want to because It will need a bit of refactoring
(change 	`settledCallback func(lntypes.Hash)` in node_settle_handler.go, mux.go,...

But I can if you think that the re-fetch is really annoying!